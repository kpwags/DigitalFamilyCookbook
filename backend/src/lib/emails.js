const requestResetPasswordEmail = (name, resetToken) => `
  <div className="email" style="
    border: 1px solid black;
    padding: 20px;
    font-family: sans-serif;
    line-height: 2;
    font-size: 20px;">
      <h2>Hello ${name},</h2>
      <p>A request was made to reset your password. Please <a href="${
  process.env.DFC_FRONTEND_URL
}/reset-password?resetToken=${resetToken}">click here to reset your password</a> or copy the following link into your browser</p>
      <p>${process.env.DFC_FRONTEND_URL}/reset-password?resetToken=${resetToken}</p>
      <p>If you did not request a reset, please disregard and delete this email.</p>
      <p>Thanks,<br/>${process.env.DFC_SITE_TITLE}</p>
  </div>
`;

exports.requestResetPasswordEmail = requestResetPasswordEmail;
