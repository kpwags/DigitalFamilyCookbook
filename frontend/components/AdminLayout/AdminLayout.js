import styled from 'styled-components';
import PropTypes from 'prop-types';
import Link from 'next/link';
import { MobileAlertBox } from '../MobileAlertBox/MobileAlertBox';

const Layout = styled.div`
    display: grid;
    grid-template-columns: 1fr 5fr;
    grid-template-rows: 1fr;
    grid-column-gap: 10px;
    grid-row-gap: 0px;

    @media all and (max-width: 800px) {
        display: block;
    }
`;

const Sidebar = styled.div`
    grid-column-start: 1;
    grid-column-end: 1;

    h3 {
        color: ${(props) => props.theme.darkGreen};
        margin-top: 0;
        padding-top: 0;
    }

    ul {
        margin: 0;
        padding: 0;
    }

    li {
        list-style-type: none;
        margin: 0 0 2rem 0;
        font-size: 1.1rem;

        a {
            color: ${(props) => props.theme.green};
        }
    }

    li.active {
        font-weight: bold;
        color: ${(props) => props.theme.darkGreen};
    }

    @media all and (max-width: 800px) {
        display: none;
    }
`;

const Main = styled.div`
    grid-column-start: 2;
    grid-column-end: 2;
`;

const AdminLayout = (props) => {
    return (
        <Layout data-test="admin-layout">
            <Sidebar>
                <h3>Site Administration</h3>
                <ul>
                    <li className={props.activePage === 'invitationcodes' ? 'active' : ''}>
                        {props.activePage === 'invitationcodes' ? (
                            <>Invitation Codes</>
                        ) : (
                            <Link href="/admin/invitation-codes">
                                <a className="gray">Invitation Codes</a>
                            </Link>
                        )}
                    </li>
                    <li className={props.activePage === 'familymembers' ? 'active' : ''}>
                        {props.activePage === 'familymembers' ? (
                            <>Family Members</>
                        ) : (
                            <Link href="/admin/users">
                                <a>Family Members</a>
                            </Link>
                        )}
                    </li>
                    <li className={props.activePage === 'categories' ? 'active' : ''}>
                        {props.activePage === 'categories' ? (
                            <>Categories</>
                        ) : (
                            <Link href="/admin/categories">
                                <a>Categories</a>
                            </Link>
                        )}
                    </li>
                    <li className={props.activePage === 'meats' ? 'active' : ''}>
                        {props.activePage === 'meats' ? (
                            <>Meats</>
                        ) : (
                            <Link href="/admin/meats">
                                <a>Meats</a>
                            </Link>
                        )}
                    </li>
                    <li className={props.activePage === 'recipes' ? 'active' : ''}>
                        {props.activePage === 'recipes' ? (
                            <>Recipes</>
                        ) : (
                            <Link href="/admin/recipes">
                                <a>Recipes</a>
                            </Link>
                        )}
                    </li>
                </ul>
            </Sidebar>
            <Main>
                <MobileAlertBox message="This is best viewed in landscape mode" />
                {props.children}
            </Main>
        </Layout>
    );
};

AdminLayout.propTypes = {
    activePage: PropTypes.string,
    children: PropTypes.node.isRequired,
};

export { AdminLayout };
