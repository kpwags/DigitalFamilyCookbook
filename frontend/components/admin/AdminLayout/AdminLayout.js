import styled from 'styled-components';
import PropTypes from 'prop-types';
import Link from 'next/link';
import { MobileAlertBox } from '../../MobileAlertBox/MobileAlertBox';

const Layout = styled.div`
    display: grid;
    grid-template-columns: 1fr 5fr;
    grid-template-rows: 1fr;
    grid-column-gap: 10px;
    grid-row-gap: 0px;

    @media all and (max-width: 800px) {
        display: block;
    }
`;

const Sidebar = styled.div`
    grid-column-start: 1;
    grid-column-end: 1;

    ul {
        margin: 0;
        padding: 0;
    }

    li {
        list-style-type: none;
        margin: 0 0 2rem 0;
        font-size: 1.2rem;

        a {
            color: ${props => props.theme.green};
        }
    }

    li.active a {
        font-weight: bold;
        color: ${props => props.theme.darkGreen};
    }

    @media all and (max-width: 800px) {
        display: none;
    }
`;

const Main = styled.div`
    grid-column-start: 2;
    grid-column-end: 2;
`;

const AdminLayout = props => {
    return (
        <Layout>
            <Sidebar>
                <ul>
                    <li className={props.activePage === 'invitationcodes' ? 'active' : ''}>
                        <Link href="/admin/invitation-codes">
                            <a className="gray">Manage Invitation Codes</a>
                        </Link>
                    </li>
                    <li className={props.activePage === 'familymembers' ? 'active' : ''}>
                        <Link href="/admin/users">
                            <a className="orange">Manage Family Members</a>
                        </Link>
                    </li>
                    <li className={props.activePage === 'categories' ? 'active' : ''}>
                        <Link href="/admin/categories">
                            <a className="green">Manage Categories</a>
                        </Link>
                    </li>
                    <li className={props.activePage === 'meats' ? 'active' : ''}>
                        <Link href="/admin/meats">
                            <a className="blue">Manage Meats</a>
                        </Link>
                    </li>
                    <li className={props.activePage === 'recipes' ? 'active' : ''}>
                        <Link href="/admin/recipes">
                            <a className="purple">Manage Recipes</a>
                        </Link>
                    </li>
                </ul>
            </Sidebar>
            <Main>
                <MobileAlertBox message="This is best viewed in landscape mode" />
                {props.children}
            </Main>
        </Layout>
    );
};

AdminLayout.propTypes = {
    activePage: PropTypes.string,
    children: PropTypes.node.isRequired
};

export { AdminLayout };
