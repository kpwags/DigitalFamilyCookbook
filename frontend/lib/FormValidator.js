import validator from 'validator';

class FormValidator {
    static validateEmail(value) {
        if (typeof value !== 'string' || !validator.isEmail(value)) {
            return false;
        }

        return true;
    }

    static validateInvitationCode(value) {
        if (!this.validateNotEmpty(value)) {
            return {
                valid: false,
                message: 'Invitation code is required'
            };
        }

        const regEx = new RegExp('^[A-Za-z0-9_]+$');
        if (!regEx.test(value)) {
            return {
                valid: false,
                message: 'Invitation codes can only contain letters, numbers, and underscores'
            };
        }

        if (value.trim().length < 1 || value.trim().length > 20) {
            return {
                valid: false,
                message: 'Invitation codes must be between 1 and 20 characters'
            };
        }

        return {
            valid: true,
            message: ''
        };
    }

    static validateNotEmpty(value) {
        if (typeof value !== 'string' || value.trim().length === 0) {
            return false;
        }

        return true;
    }

    static validateSetLength(value, fieldName = 'value', minLength = 1, maxLength = 25) {
        if (!this.validateNotEmpty(value)) {
            return {
                valid: false,
                message: `${fieldName} is required`
            };
        }

        if (value.trim().length < minLength || value.trim().length > maxLength) {
            return {
                valid: false,
                message: `${fieldName} must be between ${minLength} and ${maxLength} characters`
            };
        }

        return {
            valid: true,
            message: ''
        };
    }

    static validateUsername(value) {
        if (!this.validateNotEmpty(value)) {
            return {
                valid: false,
                message: 'Username is required'
            };
        }

        const regEx = new RegExp('^[A-Za-z0-9_]+$');
        if (!regEx.test(value)) {
            return {
                valid: false,
                message: 'Username can only contain letters, numbers, and underscores'
            };
        }

        if (value.trim().length < 1 || value.trim().length > 20) {
            return {
                valid: false,
                message: 'Username must be between 1 and 20 characters'
            };
        }

        return {
            valid: true,
            message: ''
        };
    }

    static validatePassword(value1, value2) {
        if (typeof value1 !== 'string' || typeof value2 !== 'string' || (value1 === '' && value2 === '')) {
            return {
                valid: false,
                message: 'Password is required'
            };
        }

        if (value1.trim() !== value2.trim()) {
            return {
                valid: false,
                message: 'Passwords do not match'
            };
        }

        if (value1.trim().length < 8) {
            return {
                valid: false,
                message: 'Passwords must be at least 8 characters long'
            };
        }

        return {
            valid: true,
            message: ''
        };
    }
}

export { FormValidator };
