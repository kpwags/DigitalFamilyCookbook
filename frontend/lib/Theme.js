const Theme = {
    black: 'hsl(0, 0%, 22.4%)',
    green: 'hsl(140, 100%, 25%)',
    lightGreen: 'hsl(136, 44%, 47%)',
    paleGreen: 'hsl(136, 100%, 88%)',
    darkGreen: 'hsl(140, 100%, 15%)',
    bs: '0 12px 24px 0 hsla(0, 0%, 0%, 0.2)'
};

export { Theme };
