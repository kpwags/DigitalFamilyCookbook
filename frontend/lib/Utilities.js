import Router from 'next/router';

class Utilities {
    static setTableRowActive(tableId, rowId) {
        const table = document.getElementById(tableId);
        const tbody = table.getElementsByTagName('tbody')[0];

        // TODO: Find better way?
        // eslint-disable-next-line no-restricted-syntax
        for (const row of tbody.getElementsByTagName('tr')) {
            row.className = '';
        }

        document.getElementById(rowId).className = 'selected';
    }

    // TODO: Add Message
    static invalidateField(elementId, errorMessage = '') {
        document.getElementById(elementId).className = 'errored';
        document.getElementById(elementId).parentElement.className = 'errored';

        if (errorMessage !== '') {
            document.getElementById(`${elementId}-message`).innerText = errorMessage;
            document.getElementById(`${elementId}-message`).style.display = 'block';
        }
    }

    static invalidateFieldByClass(className) {
        const elements = document.getElementsByClassName(className);

        for (let i = 0; i < elements.length; i += 1) {
            elements[i].className = `${className}  errored`;
        }
    }

    static resetFieldByClass(className) {
        const elements = document.getElementsByClassName(className);

        for (let i = 0; i < elements.length; i += 1) {
            let elementClassName = elements[i].className;
            elementClassName = elementClassName.replace('errored', '');
            elements[i].className = elementClassName;
        }
    }

    static resetField(elementId) {
        let { className } = document.getElementById(elementId);
        className = className.replace('errored', '');
        document.getElementById(elementId).className = className;

        let parentClassName = document.getElementById(elementId).parentElement.className;
        parentClassName = parentClassName.replace('errored', '');
        document.getElementById(elementId).parentElement.className = parentClassName;
        document.getElementById(`${elementId}-message`).style.display = 'none';
        document.getElementById(`${elementId}-message`).innerText = '';
    }

    static getNextAvailableValue(arr, idx = 'key') {
        let nextValue = 0;
        arr.forEach(item => {
            if (item[idx] >= nextValue) {
                nextValue = item[idx];
            }
        });

        nextValue += 1;

        return nextValue;
    }

    static convertCookingTime(time) {
        let timeHours = 0;
        let timeMinutes = 0;

        let returnString = '';

        if (time !== null && parseInt(time, 10) > 0) {
            timeHours = Math.floor(time / 60);
            timeMinutes = time % 60;

            if (timeHours === 1) {
                returnString += `${timeHours} hour`;
            } else if (timeHours > 1) {
                returnString += `${timeHours} hours`;
            }

            if (timeHours > 0 && timeMinutes > 0) {
                returnString += ', ';
            }

            if (timeMinutes > 0) {
                returnString += `${timeMinutes} minutes`;
            }
        }

        return returnString;
    }

    static convertIDArray(arr) {
        const newArr = [];
        arr.forEach(i => {
            newArr.push(i.id);
        });

        return newArr;
    }

    static toTitleCase(str) {
        return str
            .toLowerCase()
            .split(' ')
            .map(word => {
                return word.replace(word[0], word[0].toUpperCase());
            })
            .join(' ');
    }

    static goToRecipe(id) {
        Router.push({
            pathname: '/recipe',
            query: { id }
        });
    }

    static toggleElement(elementId) {
        if (document.getElementById(elementId).style.display !== 'none') {
            document.getElementById(elementId).style.display = 'none';
        } else {
            document.getElementById(elementId).style.display = 'block';
        }
    }

    static toggleMobileMenu() {
        document.getElementById('hamburgermenu').classList.toggle('change');

        if (document.getElementById('mobilemenu').style.left === '0px') {
            document.getElementById('mobilemenu').style.left = '-250px';
        } else {
            document.getElementById('mobilemenu').style.left = '0px';
        }
    }

    static toggleHeaderMenu(menuId) {
        if (
            document.getElementById(menuId).style.display === 'none' ||
            document.getElementById(menuId).style.display === ''
        ) {
            document.getElementById(menuId).style.display = 'block';
        } else {
            document.getElementById(menuId).style.display = 'none';
        }
    }

    static hideHeaderMenu(menuId) {
        document.getElementById(menuId).style.display = 'none';
    }

    static formatDate(date) {
        const formattedDate = new Date(date);
        return formattedDate.toLocaleDateString('en-US');
    }

    static boxColors = ['gray', 'blue', 'green', 'orange', 'purple'];
}

export { Utilities };
