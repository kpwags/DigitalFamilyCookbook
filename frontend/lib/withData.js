import withApollo from 'next-with-apollo';
import ApolloClient from 'apollo-boost';
import { endpoint } from '../config';
import { LOCAL_STATE_QUERY } from '../queries/Local';

function createClient({ headers }) {
    return new ApolloClient({
        uri: endpoint,
        request: operation => {
            operation.setContext({
                fetchOptions: {
                    credentials: 'include'
                },
                headers
            });
        },
        // local data
        clientState: {
            resolvers: {
                Mutation: {
                    toggleOverlay(_, variables, { cache }) {
                        const { overlayVisible } = cache.readQuery({
                            query: LOCAL_STATE_QUERY
                        });

                        const data = {
                            data: {
                                overlayVisible: !overlayVisible
                            }
                        };

                        cache.writeData(data);

                        return data;
                    }
                }
            },
            defaults: {
                overlayVisible: false
            }
        }
    });
}

export default withApollo(createClient);
