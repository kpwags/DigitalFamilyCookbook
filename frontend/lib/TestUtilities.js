/* eslint-disable import/no-extraneous-dependencies */
import casual from 'casual';
import PropTypes from 'prop-types';
import { MockedProvider } from '@apollo/react-testing';
import { ThemeProvider } from 'styled-components';
import { Theme } from './Theme';

const MockedThemeProvider = props => {
    const { mocks = null, children } = props;
    return (
        <ThemeProvider theme={Theme}>
            <MockedProvider mocks={mocks}>{children}</MockedProvider>
        </ThemeProvider>
    );
};

MockedThemeProvider.propTypes = {
    children: PropTypes.node,
    mocks: PropTypes.array
};

const TestCategory = () => ({
    __typename: 'Category',
    id: 'abc123',
    name: casual.words()
});

const TestDirection = () => ({
    __typename: 'Direction',
    id: casual.uuid,
    name: casual.words(),
    sortOrder: casual.integer
});

const TestIngredient = () => ({
    __typename: 'Ingredient',
    id: casual.uuid,
    name: casual.words(),
    sortOrder: casual.integer
});

const TestInvitationCode = () => ({
    __typename: 'InvitationCode',
    id: 'abc123',
    code: casual.username
});

const TestMeat = () => ({
    __typename: 'Meat',
    id: 'abc123',
    name: casual.words()
});

const TestRecipe = () => ({
    __typename: 'Recipe',
    id: 'abc123',
    name: casual.name,
    public: casual.boolean,
    source: casual.words(),
    sourceUrl: casual.url,
    time: casual.integer,
    activeTime: casual.integer,
    servings: casual.integer,
    calories: casual.integer,
    carbohydrates: casual.integer,
    fat: casual.integer,
    sugar: casual.integer,
    cholesterol: casual.integer,
    fiber: casual.integer,
    image: 'recipe.jpg',
    largeImage: 'recipe-large.jpg',
    ingredients: [TestIngredient(), TestIngredient()],
    directions: [TestDirection(), TestDirection()],
    meats: [TestMeat(), TestMeat()],
    categories: [TestCategory(), TestCategory()]
});

const TestRecipesList = () => [TestRecipe(), TestRecipe(), TestRecipe(), TestRecipe(), TestRecipe()];

const TestUser = () => ({
    __typename: 'User',
    id: 'abc123',
    name: casual.name,
    username: casual.username,
    email: casual.email,
    bio: 'I am just a test user',
    image: 'user.jpg',
    largeImage: 'user-lg.jpg',
    permissions: ['USER']
});

export {
    MockedThemeProvider,
    TestCategory,
    TestDirection,
    TestIngredient,
    TestInvitationCode,
    TestMeat,
    TestRecipe,
    TestRecipesList,
    TestUser
};
