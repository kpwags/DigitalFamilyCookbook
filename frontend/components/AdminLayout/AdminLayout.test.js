import { mount } from 'enzyme';
import toJson from 'enzyme-to-json';
import { MockedThemeProvider } from '../../lib/TestUtilities';
import { AdminLayout } from './AdminLayout';

describe('<AdminLayout/>', () => {
    it('should match snapshot', () => {
        const wrapper = mount(
            <MockedThemeProvider>
                <AdminLayout>
                    <div />
                </AdminLayout>
            </MockedThemeProvider>
        );

        expect(toJson(wrapper.find('div[data-test="admin-layout"]'))).toMatchSnapshot();
    });
});
