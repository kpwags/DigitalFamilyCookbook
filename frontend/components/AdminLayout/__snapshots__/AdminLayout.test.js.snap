// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<AdminLayout/> should match snapshot 1`] = `
<div
  className="AdminLayout__Layout-sc-1mt2f2w-0 fwmAWL"
  data-test="admin-layout"
>
  <AdminLayout__Sidebar>
    <StyledComponent
      forwardedComponent={
        Object {
          "$$typeof": Symbol(react.forward_ref),
          "attrs": Array [],
          "componentStyle": ComponentStyle {
            "componentId": "AdminLayout__Sidebar-sc-1mt2f2w-1",
            "isStatic": false,
            "lastClassName": "jmHxvD",
            "rules": Array [
              "grid-column-start:1;grid-column-end:1;ul{margin:0;padding:0;}li{list-style-type:none;margin:0 0 2rem 0;font-size:1.2rem;a{color:",
              [Function],
              ";}}li.active a{font-weight:bold;color:",
              [Function],
              ";}@media all and (max-width:800px){display:none;}",
            ],
          },
          "displayName": "AdminLayout__Sidebar",
          "foldedComponentIds": Array [],
          "render": [Function],
          "styledComponentId": "AdminLayout__Sidebar-sc-1mt2f2w-1",
          "target": "div",
          "toString": [Function],
          "usesTheme": true,
          "warnTooManyClasses": [Function],
          "withComponent": [Function],
        }
      }
      forwardedRef={null}
    >
      <div
        className="AdminLayout__Sidebar-sc-1mt2f2w-1 jmHxvD"
      >
        <ul>
          <li
            className=""
          >
            <Link
              href="/admin/invitation-codes"
            >
              <a
                className="gray"
                href="/admin/invitation-codes"
                onClick={[Function]}
                onMouseEnter={[Function]}
              >
                Manage Invitation Codes
              </a>
            </Link>
          </li>
          <li
            className=""
          >
            <Link
              href="/admin/users"
            >
              <a
                className="orange"
                href="/admin/users"
                onClick={[Function]}
                onMouseEnter={[Function]}
              >
                Manage Family Members
              </a>
            </Link>
          </li>
          <li
            className=""
          >
            <Link
              href="/admin/categories"
            >
              <a
                className="green"
                href="/admin/categories"
                onClick={[Function]}
                onMouseEnter={[Function]}
              >
                Manage Categories
              </a>
            </Link>
          </li>
          <li
            className=""
          >
            <Link
              href="/admin/meats"
            >
              <a
                className="blue"
                href="/admin/meats"
                onClick={[Function]}
                onMouseEnter={[Function]}
              >
                Manage Meats
              </a>
            </Link>
          </li>
          <li
            className=""
          >
            <Link
              href="/admin/recipes"
            >
              <a
                className="purple"
                href="/admin/recipes"
                onClick={[Function]}
                onMouseEnter={[Function]}
              >
                Manage Recipes
              </a>
            </Link>
          </li>
        </ul>
      </div>
    </StyledComponent>
  </AdminLayout__Sidebar>
  <AdminLayout__Main>
    <StyledComponent
      forwardedComponent={
        Object {
          "$$typeof": Symbol(react.forward_ref),
          "attrs": Array [],
          "componentStyle": ComponentStyle {
            "componentId": "AdminLayout__Main-sc-1mt2f2w-2",
            "isStatic": false,
            "lastClassName": "kwdPCQ",
            "rules": Array [
              "grid-column-start:2;grid-column-end:2;",
            ],
          },
          "displayName": "AdminLayout__Main",
          "foldedComponentIds": Array [],
          "render": [Function],
          "styledComponentId": "AdminLayout__Main-sc-1mt2f2w-2",
          "target": "div",
          "toString": [Function],
          "usesTheme": false,
          "warnTooManyClasses": [Function],
          "withComponent": [Function],
        }
      }
      forwardedRef={null}
    >
      <div
        className="AdminLayout__Main-sc-1mt2f2w-2 kwdPCQ"
      >
        <MobileAlertBox
          message="This is best viewed in landscape mode"
        >
          <MobileAlertBox__AlertBox
            id="mobile_alert_box"
          >
            <StyledComponent
              forwardedComponent={
                Object {
                  "$$typeof": Symbol(react.forward_ref),
                  "attrs": Array [],
                  "componentStyle": ComponentStyle {
                    "componentId": "MobileAlertBox__AlertBox-sc-18higoe-0",
                    "isStatic": false,
                    "lastClassName": "dcfKCe",
                    "rules": Array [
                      "display:none;background:hsl(0,0%,22%);color:hsl(100,100%,100%);position:absolute;bottom:0px;left:0;font-size:1rem;width:100%;padding:12px 8px;.close{position:absolute;top:5px;right:10px;a{color:hsl(100,100%,100%);}a:hover{text-decoration:none;}}@media all and (max-width:500px){display:block;}",
                    ],
                  },
                  "displayName": "MobileAlertBox__AlertBox",
                  "foldedComponentIds": Array [],
                  "render": [Function],
                  "styledComponentId": "MobileAlertBox__AlertBox-sc-18higoe-0",
                  "target": "div",
                  "toString": [Function],
                  "usesTheme": false,
                  "warnTooManyClasses": [Function],
                  "withComponent": [Function],
                }
              }
              forwardedRef={null}
              id="mobile_alert_box"
            >
              <div
                className="MobileAlertBox__AlertBox-sc-18higoe-0 dcfKCe"
                id="mobile_alert_box"
              >
                This is best viewed in landscape mode
                <div
                  className="close"
                >
                  <a
                    onClick={[Function]}
                    onKeyDown={[Function]}
                    role="button"
                    tabIndex="0"
                  >
                    x
                  </a>
                </div>
              </div>
            </StyledComponent>
          </MobileAlertBox__AlertBox>
        </MobileAlertBox>
        <div />
      </div>
    </StyledComponent>
  </AdminLayout__Main>
</div>
`;
